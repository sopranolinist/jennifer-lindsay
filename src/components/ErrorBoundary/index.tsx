import {
  createStyles,
  Theme,
  withStyles,
  WithStyles
} from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import React, { Component, ReactChild } from 'react';
import DocumentTitle from 'react-document-title';
import { getPageTitle } from '../../helpers/DataHelper';

const styles = (theme: Theme) =>
  createStyles({
    root: {
      alignItems: 'baseline',
      padding: theme.spacing(3),
      paddingTop: theme.spacing(12),
      flexGrow: 1,
      flexDirection: 'column',
      display: 'flex',
      textAlign: 'left'
    },
    errorHeader: {
      color: theme.palette.error['main'],
      marginBottom: theme.spacing(2)
    },
    errorMessage: {
      marginBottom: theme.spacing(2)
    }
  });

interface Props extends WithStyles<typeof styles> {
  children?: ReactChild;
  hasError?: string;
}
interface State {
  error: boolean;
}

class ErrorBoundary extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { error: false };
  }

  static getDerivedStateFromError(error: Error | null) {
    return { error: true };
  }

  componentDidCatch(error: Error | null, info: object) {
    // console.log('ErrorBoundary.componentDidCatch', error, info);
  }

  render() {
    const pageTitle = 'Error';
    const { error } = this.state;
    const { children, hasError, classes } = this.props;
    if (!error && !hasError) return children;
    return (
      <div className={classes.root}>
        <DocumentTitle title={getPageTitle(pageTitle)} />
        <Typography component="h1" variant="h6" className={classes.errorHeader}>
          Error
        </Typography>
        <Typography
          component="span"
          variant="body1"
          className={classes.errorMessage}
        >
          The site has encountered an error{hasError && ':'}
        </Typography>
        {hasError && (
          <Typography
            component="span"
            variant="body1"
            className={classes.errorMessage}
          >
            {hasError}
          </Typography>
        )}
      </div>
    );
  }
}

export default withStyles(styles)(ErrorBoundary);
