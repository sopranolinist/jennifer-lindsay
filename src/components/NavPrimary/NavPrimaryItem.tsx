import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import React, { ReactElement } from 'react';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import { isSelected } from '../../helpers/NavHelper';

interface Props extends RouteComponentProps {
  label: string;
  icon?: () => ReactElement;
  pathname?: string;
  path?: string;
  subnav?: () => void;
  external?: boolean;
  name: string;
  children?: ReactElement;
  closeDrawer: () => void;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      paddingTop: 16,
      paddingBottom: 16,
    },
  }),
);

const NavPrimaryItem = ({
  label,
  icon,
  pathname,
  path,
  subnav,
  external,
  name,
  history,
  location,
  children,
  closeDrawer,
}: Props) => {
  const classes = useStyles();

  const handleClick = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
    if (external) {
      closeDrawer();
    } else if (subnav) {
      subnav();
    } else {
      event.preventDefault();
      closeDrawer();
      history.push({
        pathname,
      });
    }
  };
  return (
    <>
      <ListItem
        button
        data-testid={`navprimaryitem-${name}`}
        selected={isSelected(path, location.pathname)}
        component="a"
        href={pathname}
        onClick={handleClick}
        className={classes.root}
        {...(external
          ? {
              target: '_blank',
            }
          : '')}
      >
        {icon && <ListItemIcon>{icon()}</ListItemIcon>}
        <ListItemText primary={label} inset={!icon} />
      </ListItem>
      {children}
    </>
  );
};

export default withRouter(NavPrimaryItem);
