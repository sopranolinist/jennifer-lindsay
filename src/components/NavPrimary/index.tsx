import Divider from '@material-ui/core/Divider';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import { createStyles, Theme, withStyles, WithStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import { format } from 'date-fns';
import React from 'react';
import { drawerWidth } from '../../helpers/NavHelper';
import {
  routeHome,
  routeContact,
} from '../../helpers/RouteHelper';
import { PollIcon } from '../../images/Icons';
import NavPrimaryItem from './NavPrimaryItem';

const styles = (theme: Theme) =>
  createStyles({
    toolbar: theme.mixins.toolbar,
    drawer: {
      [theme.breakpoints.up('sm')]: {
        width: drawerWidth,
        flexShrink: 0,
      },
    },
    drawerPaper: {
      width: drawerWidth,
    },
    authStatus: {
      padding: theme.spacing(2),
    },
    nested: {
      paddingLeft: theme.spacing(4),
    },
  });

interface Props extends WithStyles<typeof styles> {
  theme: Theme;
  mobileOpen: boolean;
  closeDrawer: () => void;
}

const NavPrimary = ({ classes, theme, mobileOpen, closeDrawer }: Props) => {
  const breakpointIsXl = useMediaQuery(theme.breakpoints.up('xl'));

  const DashIcon = () => <PollIcon />;
  return (
    <nav className={classes.drawer}>
      <Drawer
        variant={breakpointIsXl ? 'permanent' : 'temporary'}
        open={breakpointIsXl ? true : mobileOpen}
        classes={{
          paper: classes.drawerPaper,
        }}
        {...(!breakpointIsXl
          ? {
              anchor: theme.direction === 'rtl' ? 'right' : 'left',
              onClose: closeDrawer,
            }
          : {})}
      >
        <div>
          <div className={classes.toolbar} />
          <div className={classes.authStatus}>
            <Typography align="left" noWrap variant="body2" component="h4">
              {'Menu'}
            </Typography>
          </div>
          <Divider />
          <List disablePadding>
            <NavPrimaryItem
              label="Home"
              icon={DashIcon}
              path={routeHome()}
              pathname={routeHome()}
              name="home"
              key="home"
              closeDrawer={closeDrawer}
            />
            <Divider variant="inset" />
            <NavPrimaryItem
              label="Contact"
              path={routeContact()}
              pathname={routeContact()}
              name="contact"
              key="contact"
              closeDrawer={closeDrawer}
            />
          </List>
        </div>
      </Drawer>
    </nav>
  );
};

export default withStyles(styles, { withTheme: true })(NavPrimary);
