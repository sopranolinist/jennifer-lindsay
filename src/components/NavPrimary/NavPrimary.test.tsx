import { render } from '@testing-library/react';
import React from 'react';
import {
  authExamples,
  getFullyWrappedComponent,
  getJssWrappedComponent,
  getRouterWrappedComponent,
  mockConsole
} from '../../helpers/TestHelper';
import ErrorBoundary from '../ErrorBoundary';
import NavPrimary from './';

jest.mock('@material-ui/core/useMediaQuery', () => jest.fn());

describe('Component - NavPrimary', () => {
  beforeEach(mockConsole);

  test('should enforce auth context', async done => {
    const mobileOpen = true;
    const closeDrawer = () => {};
    const { container } = render(
      getJssWrappedComponent(
        getRouterWrappedComponent(
          <ErrorBoundary>
            <NavPrimary mobileOpen={mobileOpen} closeDrawer={closeDrawer} />
          </ErrorBoundary>
        )
      )
    );
    expect(container).toMatchSnapshot();
    done();
  });

  test('renders the NavPrimary screen', async done => {
    const { authContextProps, authObjProps } = authExamples.signedInAuthorized;
    const mobileOpen = true;
    const closeDrawer = () => {};
    const { container } = render(
      getFullyWrappedComponent(
        <NavPrimary mobileOpen={mobileOpen} closeDrawer={closeDrawer} />,
        {
          authContextProps,
          authObjProps
        }
      )
    );
    expect(container).toMatchSnapshot();
    done();
  });

  test('renders empty NavPrimary screen for unauthenticated users', async done => {
    const mobileOpen = true;
    const closeDrawer = () => {};
    const { container } = render(
      getFullyWrappedComponent(
        <NavPrimary mobileOpen={mobileOpen} closeDrawer={closeDrawer} />
      )
    );
    expect(container).toMatchSnapshot();
    done();
  });
});
