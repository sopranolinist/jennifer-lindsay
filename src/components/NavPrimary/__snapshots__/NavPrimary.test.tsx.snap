// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - NavPrimary renders empty NavPrimary screen for unauthenticated users 1`] = `
<div
  aria-hidden="true"
>
  <nav
    class="NavPrimary-drawer-5"
  />
</div>
`;

exports[`Component - NavPrimary renders the NavPrimary screen 1`] = `
<div
  aria-hidden="true"
>
  <nav
    class="NavPrimary-drawer-5"
  />
</div>
`;

exports[`Component - NavPrimary should enforce auth context 1`] = `
<div>
  <div
    class="ErrorBoundary-root-1"
  >
    <h1
      class="MuiTypography-root ErrorBoundary-errorHeader-2 MuiTypography-h6"
    >
      Error
    </h1>
    <span
      class="MuiTypography-root ErrorBoundary-errorMessage-3 MuiTypography-body1"
    >
      The site has encountered an error
    </span>
  </div>
</div>
`;
